File: Backend/Entities/CourseCoordinatorInstructor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Entities
{
    /// <summary>A class for a course coordinator or instructor of a course.</summary>
    /// <remarks>
    /// A course coordinator is a user who is responsible for a course.
    /// </remarks>
    [Table("CourseCoordinatorInstructors")]
    public class CourseCoordinatorInstructor : DomainUser
    {
        public DepartmentInfo Department { get; set; }

        // TODO: Convert this to a CourseList
        public Course Course { get; set; }
        public bool IsCourseCoordinator { get; set; } = false;
    }
}

File: Frontend/src/app/_models/course-coordinator-instructor.ts
import { DepartmentInfo } from './department-info';
import { Course } from './course';
import { DomainUser } from './domain-user';

export interface CourseCoordinatorInstructor extends DomainUser {
  department: DepartmentInfo;
  course: Course;
  isCourseCoordinator: boolean;
}
